### PR TITLE
fix: ignore retry node when check succeeded descendant nodes. Fixes: #13003

### LINKED ISSUE #13003: Workflows that failed before upgrade to 3.5.6 fail to retry
### Pre-requisites

- [X] I have double-checked my configuration
- [X] I have tested with the `:latest` image tag (i.e. `quay.io/argoproj/workflow-controller:latest`) and can confirm the issue still exists on `:latest`. If not, I have explained why, **in detail**, in my description below.
- [X] I have searched existing issues and could not find a match for this bug
- [ ] I'd like to contribute the fix myself (see [contributing guide](https://github.com/argoproj/argo-workflows/blob/main/docs/CONTRIBUTING.md))

### What happened/what did you expect to happen?

1. We were on 3.5.5
1. Some workflows failed
1. Upgraded to 3.5.6
1. Retried some of the workflows
Result:

    - The failed tasks within the workflows disappeared, keeping only the successful ones
    - The workflows got stuck in a running state
    - Still The workflows are shown in the failed list
    - Unable to retry again, because it's running
    - Unable to Stop 

Reproducible on any workflow

### Version

v3.5.6

### Paste a small workflow that reproduces the issue. We must be able to run the workflow; don't enter a workflows that uses private images.

```YAML
any workflow reproduces it
```

### Logs from the workflow controller

```text
kubectl logs -n argo deploy/workflow-controller | grep ${workflow}
```

### Logs from in your workflow's wait container

```text
kubectl logs -n argo -c wait -l workflows.argoproj.io/workflow=${workflow},workflow.argoproj.io/phase!=Succeeded
```

### PR DESCRIPTION
**Pull request by shuangkun**
ignore retry node when check descendant nodes.

Fixes #13003

### CODE REVIEW

File: workflow/util/util.go [hunk 1]
```diff
@@ -971,7 +971,7 @@ func FormulateRetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSucce
 				log.Debugf("Reset %s node %s since it's a group node", node.Name, string(node.Phase))
 				continue
 			} else {
-				if isDescendantNodeSucceeded(wf, node, nodeIDsToReset) {
+				if node.Type == wfv1.NodeTypePod && isDescendantNodeSucceeded(wf, node, nodeIDsToReset) {
```
Comment (Joibel) [inline workflow/util/util.go:974]: It feels like this isn't the whole list of NodeTypes that should be considered here.
HTTP, Plugin and Container come to mind, but I haven't thought it through completely.

**Review by Joibel — CHANGES_REQUESTED** [review]

File: workflow/util/util.go (hunk 1 above)
Comment (shuangkun) [reply workflow/util/util.go]: Oh, I originally wrote it as not equal to retry, but there was a test error, and now I changed it back.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**